src/path.jl

```diff
@@ -144,6 +132,39 @@ function parents(path::T) where {T <: AbstractPath}
     end
 end
 
+"""
+  *(a::T, b::Union{T, AbstractString, AbstractChar}...) where {T <: AbstractPath} -> T
+
+Concatenate paths, strings and/or characters, producing a new path.
+This is equivalent to concatenating the string representations of paths and other strings
+and then constructing a new path.
+
+# Example
+
+julia> p"foo" * "bar"
+p"foobar"
+"""
+function Base.:(*)(a::T, b::Union{T, AbstractString, AbstractChar}...) where T <: AbstractPath
+    T(*(string(a), string.(b)...))
+end
+
+"""
+  /(a::AbstractPath, b::Union{AbstractPath, AbstractString}...) -> AbstractPath
+
+Join the path components into a new fulll path, equivalent to calling `joinpath`
+
+# Example
+
+julia> p"foo" / "bar"
+p"foo/bar"
+
+julia> p"foo" / "bar" / "baz"
+p"foo/bar/baz"
```

Comment:
Just curious what happens if you do: `p"/foo" / "/bar"`

Reply:
```
julia> p"/foo" / "/bar"
p"/bar"
```

Reply:
That's the same behaviour as base.

```
julia> joinpath("/foo", "/bar")
"/bar"
```

Reply:
Ok, thanks!